tests/test_utils.py

```diff
@@ -38,6 +38,14 @@ def test_generate_candidate_blocks_from_csv_exception(self):
         with self.assertRaises(ValueError):
             generate_candidate_blocks_from_csv(input_f, input_f, True)
 
+    def test_generate_candidate_blocks_from_csv(self):
```

Comment:
that's not a good name for the function, as you don't actually test if blocks where generated and/or correct. 
What you do test is the metadata. Why not call it `test_metadata_in_result` or something like that.

Reply:
done